src/aal/aal_cheri_mips.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#define SNMALLOC_VA_BITS_64
+
+extern "C"
+{
+#include <machine/cheri.h>
+}
+
+namespace snmalloc
+{
+  /**
+   * CHERI-MIPS specific architecture abstraction layer.
+   */
+  class AAL_CHERI_MIPS
```

Comment:
None of this looks CHERI specific.  Should this be a MIPS or BERI AAL?

Reply:
That's fair.  Though if rederivation is going to move into the AAL, that's certainly going to be CHERI-specific.  I suppose having MIPS from which CHERI-MIPS inherits is fine.